### ISSUE TITLE
Symlinks from input artifacts are not loaded

### ISSUE THREAD

**Issue by cal-pratt** [argoproj/argo-workflows#10048]
### Pre-requisites

- [X] I have double-checked my configuration
- [ ] I can confirm the issues exists when I tested with `:latest`
- [ ] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/master/docs/CONTRIBUTING.md))

### What happened/what you expected to happen?

When loading an input artifact from a `.tgz` file, symlinks from the `.tgz` file are not placed in the filesystem. This is causing my workflows to break. I'm not sure if this is new or expected behavior, but it definitely caught me off guard.

In my example I have one workflow run:
```
mkdir /tmp/test && cd /tmp/test;
echo "This is a test" > foo;
ln -s foo bar;
cat foo;
cat bar;
```
Outputs
```
This is a test
This is a test
```

I then set `/tmp/test` as an output artifact with default archive policy (.tgz) and use it as the input for my next container.

```
cd /tmp/test;
cat foo;
cat bar;
```

Outputs:
```
This is a test
cat: bar: No such file or directory
```

I can confirm that uploading the artifact is not an issue, as after downloading the tgz from the workflow I do see the file present:

```
$ tar -tf Downloads/test.tgz
test
test/bar
test/foo
```

I don't have direct access to a lot of the internal logs (as I'm using an Argo as a Service offering from one of our internal teams). Hopefully this is easily reproducible, but do let me know if you need additional details and I'll follow up with the right people on my side.

### Version

v3.4.3

### Paste a small workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

```YAML
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: test-workflow-
spec:
  entrypoint: entry
  serviceAccountName: workflow
  templates:
    - name: entry
      steps:
        - - name: create
            template: create
        - - name: load
            template: load
            arguments:
              artifacts:
                - name: test
                  from: "{{ steps.create.outputs.artifacts.test }}"
    - name: create
      outputs:
        artifacts:
          - name: test
            path: /tmp/test
      container:
        args:
          - |
            mkdir /tmp/test && cd /tmp/test;
            echo "This is a test" > foo;
            ln -s foo bar;
            cat foo;
            cat bar;
        command:
          - bash
          - -c
        image: 'docker/whalesay'
    - name: load
      inputs:
        artifacts:
          - name: test
            path: /tmp/test
      container:
        args:
          - |
            cd /tmp/test;
            cat foo;
            cat bar;
        command:
          - bash
          - -c
        image: 'docker/whalesay'
```

### Logs from the workflow controller

kubectl logs -n argo deploy/workflow-controller | grep ${workflow}

### Logs from in your workflow's wait container

kubectl logs -n argo -c wait -l workflows.argoproj.io/workflow=${workflow},workflow.argoproj.io/phase!=Succeeded

**Comment by cal-pratt** [comment]
Note: I also saw this behavior loading an input artifact that I uploaded to my s3 store outside of argo. e.g. 

```yaml
  - name: mytemplate
    inputs:
      artifacts:
      - name: mydata
        path: /some/path
        s3:
          endpoint: my.s3.endpoint
          bucket: my-bucket
          key: mydata
          accessKeySecret:
            name: s3-credentials
            key: accessKey
          secretKeySecret:
            name: s3-credentials
            key: secretKey
```

**Comment by terrytangyuan** [comment]
This should've been fixed by https://github.com/argoproj/argo-workflows/issues/9948

**Comment by cal-pratt** [comment]
> This should've been fixed by #9948

Any version info you'd like me to find out? (was reporting `v3.4.3` as that is what I see displayed in the UI).

**Comment by caelan-io** [comment]
Closing as fixed per comment above:
> This should've been fixed by https://github.com/argoproj/argo-workflows/pull/9949. Related https://github.com/argoproj/argo-workflows/issues/9948
